student-preferred-name-portlet-api/src/main/java/edu/wisc/portlet/preferred/form/validator/PreferredNameValidator.java

```diff
@@ -31,13 +38,33 @@ public void validate(Object target, Errors errors) {
 			errors.rejectValue("middleName", "error.toolong");
 		}
 		
+		if(!StringUtils.isEmpty(pn.getLastName()) && pn.getLastName().length() > 30) {
+            errors.rejectValue("lastName", "error.toolong");
+        }
+		
 		final String regx = "^[A-Za-z .-]*$";
 		Pattern ptrn = Pattern.compile(regx);
 		Matcher fnameMatcher = ptrn.matcher(pn.getFirstName());
 		Matcher mnameMatcher = ptrn.matcher(pn.getMiddleName());
 		
-		if(!fnameMatcher.find() || !mnameMatcher.find()) {
+		final String lnameregx = "^[A-Za-z .-\\\\']*$";
+        Pattern lnameptrn = Pattern.compile(lnameregx);
+        Matcher lnameMatcher = lnameptrn.matcher(pn.getLastName());
+		
+		if(!fnameMatcher.find() || !mnameMatcher.find() || !lnameMatcher.find()) {
 			errors.rejectValue("firstName", "error.invalidCharacter");
```

Comment:
If you look at the JSP, it doesn't really matter. All the error messages are in the same spot. Just did this so you don't get the same message 3 times in case you messed up on all your names.